File: LeetSync/9-palindrome-number/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x<0) return false;
    let ans = 0;
    let n = x;
    while(n>0){
        ans = ans*10 + n%10;
        n = Math.floor(n/=10)
    }
    return x==ans
};

File: LeetSync/9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int ans = 0;
        int temp = x;
        while(x!=0){
            int lastDigit = x%10;
            ans = ans*10 + lastDigit;
            x/=10;
        }
       
        return ans==temp;
    }
}